taskcluster/adhoc_taskgraph/transforms/mac_notarize.py

```diff
@@ -0,0 +1,45 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+"""
+Create signing tasks.
+"""
+from taskgraph.transforms.base import TransformSequence
+
+transforms = TransformSequence()
+
+
+@transforms.add
+def build_notarize_task(config, tasks):
+    for task in tasks:
+        if config.params["level"] != "3":
+            # Notarization only runs on level 3
+            continue
```

Comment:
As per our chat, this could be replaced with `run-on-github-branches` in the task definition. That way it will be filtered at the target tasks phase instead of the full phase (which is helpful for debugging).

Reply:
👍  TIL!
Thanks